Spec/RealtimeClientConnection.swift

```diff
@@ -1429,6 +1429,7 @@ class RealtimeClientConnection: QuickSpec {
                     expect(lastSerial).to(equal(4))
 
                     options.recover = client.connection.recoveryKey
+                    client.onError(AblyTests.newErrorProtocolMessage())
```

Comment:
@SimonWoolf @ricardopereira I assume you two have discussed this.  Shouldn't a connection that fails to resume just start a new connection?  The comment above indicates that we should put the connection into the Failed state, but that seems wrong given failed is terminal.

Reply:
@mattheworiordan the test constructs a new, second client library instance, which recovers from the first. It is not valid to recover from a connected, active client library instance -- when the new one connects the old gets disconnected as it's been superseded, so it tries to reconnect and steals the connection back from the new, and so on. The two client lib instances fight over the connection, trading it back and forth.

Reply:
Sure, I get that.  But if a client has its connection disconnected because it's been taken over, it shouldn't become `FAILED`, it should lose continuity and start a new connection.  I don't agree that's a terminal condition.

Reply:
Also, if this behaviour is expected in other libs, shouldn't this be in the spec?

Reply:
> if a client has its connection disconnected because it's been taken over, it shouldn't become FAILED, it should lose continuity and start a new connection. I don't agree that's a terminal condition.

I don't follow your objection. Being superseded doesn't trigger it to become failed, it's becoming failed because Ricardo is explicitly calling onError to make it do so, as an easy way of making it disconnect all transports and stop talking to realtime. The point is to simulate the real-life situation when recover would be used (where the previous client lib instance no longer exists because iOS has killed it or whatever).

You [do the same thing in ably-ruby tests](https://github.com/ably/ably-ruby/blob/f7a8756a27e85e1a43c92a4d4632344a83fadaef/spec/acceptance/realtime/connection_spec.rb#L1223)
I [do a similar thing in ably-js tests (but force suspended rather than failed)](https://github.com/ably/ably-js/blob/33f4b69a50ca7b1dc25241ac7780908fdd2ed78d/spec/realtime/resume.test.js#L415).

> Also, if this behaviour is expected in other libs, shouldn't this be in the spec?

I'm not clear on what you're saying should be in the spec?